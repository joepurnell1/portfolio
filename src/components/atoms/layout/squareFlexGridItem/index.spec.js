import React from 'react';
import renderer from 'react-test-renderer';
import SquareFlexGridItem from './index';

describe('Square Flex Item component', () => {
  it('should render default as expected', () => {
    const render = renderer.create(
      <SquareFlexGridItem>Today I missed my work out</SquareFlexGridItem>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with focussed prop as expected', () => {
    const render = renderer.create(
      <SquareFlexGridItem focused>But it worked out</SquareFlexGridItem>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with given size expected', () => {
    const render = renderer.create(
      <SquareFlexGridItem size="300px">I know that my girls trust is a luxury</SquareFlexGridItem>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with custom box shadow as expected', () => {
    const render = renderer.create(
      <SquareFlexGridItem size="300px">I don&apost want my next album sounding all Usher-y</SquareFlexGridItem>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with custom margin as expected', () => {
    const render = renderer.create(
      <SquareFlexGridItem margin="30px">But I must confess, I must confess</SquareFlexGridItem>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
