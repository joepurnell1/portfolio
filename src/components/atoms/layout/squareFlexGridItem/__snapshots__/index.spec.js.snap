// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Flex Item component should render default as expected 1`] = `
<div
  className="sc-bdVaJa jnbCuE"
  size="100px"
>
  Today I missed my work out
</div>
`;

exports[`Square Flex Item component should render with custom box shadow as expected 1`] = `
<div
  className="sc-bdVaJa EZUpV"
  size="100px"
>
  I don&apost want my next album sounding all Usher-y
</div>
`;

exports[`Square Flex Item component should render with custom margin as expected 1`] = `
<div
  className="sc-bdVaJa dTSNHn"
  size="100px"
>
  But I must confess, I must confess
</div>
`;

exports[`Square Flex Item component should render with focussed prop as expected 1`] = `
<div
  className="sc-bdVaJa dwOKVq"
  size="100px"
>
  But it worked out
</div>
`;

exports[`Square Flex Item component should render with given size expected 1`] = `
<div
  className="sc-bdVaJa iFlcxP"
  size="300px"
>
  I know that my girls trust is a luxury
</div>
`;
